internal/certs/certs.go

```diff
@@ -0,0 +1,153 @@
+package certs
+
+import (
+	"crypto/x509/pkix"
+	"fmt"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/pkg/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"path/filepath"
+	"time"
+)
+
+type CertManager struct {
+	CertificatesDir string
+	userCert        []byte
+	userKey         []byte
+	caBundle        []byte
+}
+type CertInfo struct {
+	keyPath         string
+	certificatePath string
+	commonName      string
+}
+
+func (r *CertManager) GenerateAllCertificates() error {
+	err := r.generateSignerCerts()
+	if err != nil {
+		return err
+	}
+	adminCA := CertInfo{
+		commonName:      "admin-kubeconfig-signer",
+		certificatePath: filepath.Join(r.CertificatesDir, "admin-kubeconfig-client-ca.crt"),
+		keyPath:         filepath.Join(r.CertificatesDir, "admin-kubeconfig-client-ca.key"),
+	}
+	ca, err := generateAndWriteCA(adminCA)
+	if err != nil {
+		return err
+	}
+
+	r.userCert, r.userKey, err = generateAdminUserCertificate(ca)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Create signer keys and certificates and return a CA bundle from all CA certificates
+func (r *CertManager) generateSignerCerts() error {
+	var certAuthData []byte
+	for _, ci := range []CertInfo{
+		{
+			keyPath:    filepath.Join(r.CertificatesDir, "loadbalancer-serving-signer.key"),
+			commonName: "kube-apiserver-lb-signer",
+		},
+		{
+			keyPath:    filepath.Join(r.CertificatesDir, "localhost-serving-signer.key"),
+			commonName: "kube-apiserver-localhost-signer",
+		},
+		{
+			keyPath:    filepath.Join(r.CertificatesDir, "service-network-serving-signer.key"),
+			commonName: "kube-apiserver-service-network-signer",
+		},
+		{
+			keyPath:    filepath.Join(r.CertificatesDir, "ingresskey-ingress-operator.key"),
+			commonName: fmt.Sprintf("%s@%d", "ingress-operator", time.Now().Unix()),
+		},
```

Comment:
I suggest we separate all the kube-apiserver certs from the ingress one, they're all bundled in the kubeconfig together but they're rather different logically